eppo_core/src/event_ingestion/event.rs

```diff
@@ -1,11 +1,39 @@
 use crate::timestamp::Timestamp;
-use serde::{Deserialize, Serialize};
+use chrono::{DateTime, Utc};
+use serde::de::Deserializer;
+use serde::{Deserialize, Serialize, Serializer};
 
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
 pub(super) struct Event {
     pub uuid: uuid::Uuid,
+    #[serde(
+        serialize_with = "serialize_millis",
+        deserialize_with = "deserialize_millis"
+    )]
     pub timestamp: Timestamp,
```

Comment:
minor: we have `serde_with` in dependencies, so we can let it do the conversions:
```suggestion
#[serde_with::serde_as]
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
pub(super) struct Event {
    pub uuid: uuid::Uuid,
    #[serde_as(as = "serde_with::TimestampMilliSeconds")]
    pub timestamp: Timestamp,
```